### ISSUE TITLE
[a11y]: Progress Bar accessibility challenges

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#11420]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

Carbon on storybook https://react.carbondesignsystem.com/?path=/story/components-progressbar--indeterminate

### React version

https://react.carbondesignsystem.com/?path=/story/components-progressbar--indeterminate

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS

### Description

Both the indeterminate and determinate versions of the Progress Bar have challenges with a screen reader.

In react, the Indeterminate progress bar does not have good JAWS output.
<img width="1706" alt="image" src="https://media.github.ibm.com/user/1900/files/4021c200-d4eb-11ec-87b3-a5b87418241d">

First, it is not announced as a live region (at least in the implementation), which results in a failure of Status Messages.

If navigating by arrow key, for each of the 3 lines, it reads:
> Progress bar label
> 0 Progress bar partially checked
> Optional helper text

The first and last lines are fine, however "0 Progress bar partially checked" is not just unhelpful but is actually confusing. Since the indeterminate is never going to resolve to another number, the 0 is meaningless. "Partially checked" is not a state I would associate with  a progress bar. (Note that a [similar output message](https://github.com/nvaccess/nvda/issues/10644) is reported with NVDA. This issue suggests it's only occurring in Chrome, but although I got an improved message in FF "50 progress bar partially checked" it's the same basic issue.)
Reviewing the [ARIA specification](https://www.w3.org/TR/wai-aria-1.2/#progressbar), the issue may be that the div with the role of progressbar has not been given `aria-busy="true"`. With this set, the progress bar is entirely bypassed and JAWS reads only the label and optional text, which is arguably a better experience.
When the loading completes, the aria-busy should be set to "false". I did not have success using any of the aria-value attribute variants (max, now, text), so the best bet may be to just convey the results via the optional helper text.

---

The [progress bar example](https://react.carbondesignsystem.com/?path=/story/components-progressbar--example) also has challenges with meaningful status info.
![image](https://media.github.ibm.com/user/1900/files/6e140000-d505-11ec-9dfd-176d00fb46d4)

The output is variable, but something like this (after a refresh):

> Export data
> 0 Progress bar partially checked
> Fetching assets...
> 28.4MB
> 490.588

Basically, the load value (in MBs) is announced once or (as above) twice, with the same odd 'partially checked' info. Here, despite the text update, there is never an announcement of "Done". 
<img width="1224" alt="image" src="https://media.github.ibm.com/user/1900/files/10cc7e80-d506-11ec-9dc4-d432fefc959c">

The code is properly using aria-value, but the output is not very successful.
`<div class="cds--progress-bar__track" role="progressbar" aria-labelledby="progress-bar-18" aria-describedby="progress-bar-helper-19" aria-valuemin="0" aria-valuemax="728" aria-valuenow="728"><div class="cds--progress-bar__bar" style="transform: scaleX(1);"></div></div>`

The 2 key things that need to be conveyed to a screen reader are 1) that the download (or whatever) is taking progress and 2) that it is done. As well, interim information on progress is welcome, if not too verbose.

It's possible that putting an `aria-live="polite"` on the following div (that contains the content of the optional text) would resolve this. Although this text is optional, my sense is that it is a rare progress bar that wouldn't expose information.

`<div id="progress-bar-helper-27" class="cds--progress-bar__helper-text">Optional helper text</div>`

### WCAG 2.1 Violation

https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html

### Reproduction/example

See storybook

### Steps to reproduce

1. Go to the storybook implementation examples
2. Launch JAWS
3. Listen to output

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
